programs/survival/signs_for_water_streams.sc

```diff
@@ -1,52 +1,57 @@
 // signs_for_water_streams by indoorjetpacks
-
+// updated for scarpet 1.7 by MeeniMc
 // Sneak click while placing signs to place it without getting the 'Edit sign message' prompt. Sneak click already skips GUI/interactions for other blocks, why not signs also? Makes sense to me.
-
 // Useful for placing water/lava flows if you make a lot of them. Lookin' at you, SciCraft server.
-
-// written for scarpet 1.6 in carpet mod 1.3.0. Both from the amazing gnembon (although...how'd you get this far without knowing that?)
-
 // tested in Carpet 1.4.24
-
 // https://github.com/gnembon/fabric-carpet/releases
-
+//written for scarpet 1.7 in carpet mod 1.4.15+. Both from the amazing gnembon (although...how'd you get this far without knowing that?)
 // namespaces for items and placed signs differ, so map each wall_sign to it's item equivalent
 
 __config() -> {
-	'stay_loaded' -> true
+	'stay_loaded' -> true,
+	'scope'->'global'
 };
-
+// Namespaces for items and placed signs differ, so map each wall_sign to it's item equivalent.
+//
+// This version uses regex matching to detect sign items, so it has a chance of supporting datapacks/mods out-of-the-box.
+// This may however cause problems where non-sign items are detected as signs and exhibit odd behavior in modded MC.
+// The original code based on an explicit whitelist remains available in comments and can be reactivated below.
 global_item_to_block = {
-	'acacia_sign'-> 'acacia_wall_sign',
-	'birch_sign'-> 'birch_wall_sign',
-	'dark_oak_sign'-> 'dark_oak_wall_sign',
-	'jungle_sign'-> 'jungle_wall_sign',
-	'oak_sign'-> 'oak_wall_sign',
-	'spruce_sign'-> 'spruce_wall_sign',
-	'crimson_sign'-> 'crimson_wall_sign',
-	'warped_sign'-> 'warped_wall_sign'
+  'acacia_sign'-> 'acacia_wall_sign',

```

Comment:
Looking good, you forgot to add the new 1.19 sign to this list though

Reply:
I will, tho if you look this map isn't actually used anywhere (it's one usage is commented out). I don't know why, cos I didn't write that code.